File: AtlasWeb/Core/Modules/Tibbi_Surec/Ketem_Modulu/CigaretteExaminationFormViewModel.cs
//$2F44165F
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class CigaretteExaminationServiceController
    {
        partial void PreScript_CigaretteExaminationForm(CigaretteExaminationFormViewModel viewModel, CigaretteExamination cigaretteExamination, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._CigaretteExamination.SubEpisode = episodeAction.SubEpisode;

            }

            ContextToViewModel(viewModel, objectContext);
        }

    }
}

namespace Core.Models
{
    public partial class CigaretteExaminationFormViewModel: BaseViewModel
    {
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Ketem_Modulu/CigaretteExaminationFormViewModel.ts
//$F0A598CB
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { CigaretteExamination } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class CigaretteExaminationFormViewModel extends BaseViewModel {
    @Type(() => CigaretteExamination)
    public _CigaretteExamination: CigaretteExamination = new CigaretteExamination();
}
